### ISSUE TITLE
Pad by >`N - 1` == NaN?

### ISSUE THREAD

**Issue by OverLordGoldDragon** [kymatio/kymatio#733]
I don't understand

```
# to avoid padding more than N - 1 on the left and on the right,
# since otherwise torch sends nans
```

I'm not aware of any NaNs in padding or `fft` by more than a certain amount, what's this about? It's needed for JTFS, and `Scattering1D` with large `T`.

@janden @MuawizChaudhary

**Comment by lostanlen** [comment]
https://github.com/kymatio/kymatio/commit/b30590248ac341f2beabeda96c9da75a804cb12e

**Comment by OverLordGoldDragon** [comment]
It's why I tagged them, but the commit provides no further info.

**Comment by OverLordGoldDragon** [comment]
I'll proceed without this restriction - even if it's still relevant, there are workarounds. Commit dates 17 months, may be early torch thing, following passes on 1.7 and 1.8:

```python
import torch
from torch.fft import fft, ifft

x  = torch.randn(128)
xp = torch.nn.functional.pad(x, (256, 256))
xpf  = fft(xp)
xpif = ifft(xp)

for arr in (xp, xpf, xpif):
    assert not bool(torch.any(torch.isnan(arr)))
```

**Comment by OverLordGoldDragon** [comment]
`RuntimeError` with

```python
x  = torch.randn(1, 1, 128)
xp = torch.nn.functional.pad(x, (256, 256), mode='reflect')
```
Then we just implement it manually. Though it's rather sketchy with `reflect`, we impose that the signal repeats itself multiple times, a nontrivial prior; even with `unpad`, this may have unintended side effects on some input kinds. But not an issue with `zero`.

**Comment by OverLordGoldDragon** [comment]
In fact @kymatio could implement all of numpy's pad kinds in a fully agnostic manner via `backend.zeros`; it's just slice and indexing manipulations on input. There's not much performance difference either, we're just writing to an array (even if there was, currently only `x` is padded, which is nothing compared to scattering).
